mijia-homie/src/main.rs

```diff
@@ -484,13 +527,14 @@ async fn handle_bluetooth_event(
     let sensors = &mut state.sensors;
     match event {
         MijiaEvent::Readings { id, readings } => {
-            if let Some(sensor) = sensors.get_mut(&id) {
+            if let Some(sensor) = get_mut_sensor_by_id(sensors, &id) {
                 sensor.publish_readings(homie, &readings).await?;
                 match sensor.connection_status {
-                    ConnectionStatus::Connected | ConnectionStatus::Connecting { .. } => {}
+                    // TODO: What should we do if the id doesn't match?
+                    ConnectionStatus::Connected { .. } | ConnectionStatus::Connecting { .. } => {}
```

Comment:
If the adaptor id doesn't match then it's still fine I think. In the worst case, that's the last one, and the watchdog timeout will kick in eventually. In the best case, we keep receiving readings from the wrong interface and don't care other than for debugging

Reply:
I think that makes sense. I'll log it, but otherwise do nothing.